File: Python/0219ContainsDuplicateII.py
def containsNearbyDuplicate(nums : list[int], k : int) -> bool:
    seenWindow = set()

    for i in range(len(nums)):
        if nums[i] in seenWindow:
            return True
        
        seenWindow.add(nums[i])

        if len(seenWindow) > k:
            seenWindow.discard(nums[i - k])
    
    return False

nums = [1,2,3,1]
k = 3

print(containsNearbyDuplicate(nums, k))

File: CPP/0219ContainsDuplicateII.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

bool containsNearbyDuplicate(vector<int>& nums, int k) {
    unordered_set<int> seenWindow;

    for (int i = 0; i < nums.size(); i++) {
        if (seenWindow.count(nums[i])) {
            return true;
        }
        seenWindow.insert(nums[i]);

        if (seenWindow.size() > k) {
            seenWindow.erase(nums[i - k]);
        }
    }

    return false;
}

int main() {
    vector<int> nums = {1, 2, 3, 1};
    int k = 3;

    cout << containsNearbyDuplicate(nums, k) << endl;
    return 0;
}